### DISCUSSION TITLE
[Spaceship] Need advice for extending spaceship features for creating IAPs

### DISCUSSION THREAD

**Discussion by Axolotl778** [fastlane/fastlane#18519] [Q&A]
I am currently trying to automate IAP creation and decided to use *spaceship*. What I'm trying to achieve is a script that uses IAP data from csv file and creates products automatically. It currently creates consumable and non-consumable IAPs flawlessly, but there is an issue with subscriptions: trial periods simply won't be added when the rest of subscription is created. I assume that it is happening because AppStore did change the way you create trial periods -- they are called introductory offers now and the web inerface is different, and *spaceship* functionality was implemented before that.

I'd like to reimplement this feature manually, but I'm having hard time understanding what kind of AppStore API *spaceship* uses to create IAPs and where I can find any docs about that.

**Comment by dinsen** [comment]
Hi @Axolotl778 

This will properly help you to find the changes in the API:
https://github.com/fastlane/fastlane/blob/master/ToolsAndDebugging.md#debugging-and-patching-spaceship-problems
